File: BE/src/main/java/com/hr_management/hr_management/entity/Role.java
package com.hr_management.hr_management.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Role")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "role_id")
    private Integer roleId;

    @Column(name = "name", nullable = false, unique = true, length = 100)
    private String name;

    @ManyToMany
    @JoinTable(
            name = "Role_Permission",
            joinColumns = @JoinColumn(name = "role_id"),
            inverseJoinColumns = @JoinColumn(name = "permission_id")
    )
    private Set<Permission> permissions = new HashSet<>();
}

File: BE-dotnet/Models/Role.cs
using System.Collections.Generic;

namespace EmpowerHR.Models
{
    public class Role
    {
        public int RoleId { get; set; }
        public string Name { get; set; }

        // Navigation Properties
        public ICollection<Permission> Permissions { get; set; } = new List<Permission>();
    }
}